### DISCUSSION TITLE
Drop the file extension from task files

### DISCUSSION THREAD

**Discussion by syhol** [jdx/mise#2249] [Ideas]
Most of my task or script files end in file extensions, .py, .sh, .js, .php, ... It would be nice if mise dropped these extensions from the 
task names.

I like to leave the extensions on the files to make it easier for me and my IDE to identify what type of file I'm working with, but when I'm calling a script, the type of file isn't important and I would much rather call `my-script` than `my-script.sh`

**Comment by hverlin** [comment]
```sh
mise tasks --json
[
  {
    "name": "build",
    "description": "Run build.sh",
    "source": "~/example-app/mise-tasks/build.sh"
  }
]
```

I think it's fixed now?

![image](https://github.com/user-attachments/assets/46dfb9f5-11be-4066-9b48-a8dbbd45ff9c)

**Reply by syhol** [reply]
This is now the the case in most places

### ✅ Mise tasks --json

<img width="430" alt="image" src="https://github.com/user-attachments/assets/c4db22f3-2dec-403c-94ae-e9eee405e6f4" />

### ✅ Mise run

<img width="521" alt="image" src="https://github.com/user-attachments/assets/2a81f58c-25fb-453e-b446-06446c1e6585" />

### ❌ But not autocomplete

<img width="430" alt="image" src="https://github.com/user-attachments/assets/ad8efb1d-6ce0-4b71-a540-e3109052614d" />

**Reply by jdx** [reply]
Are you sure it's not cached? Autocomplete just calls `mise tasks --json` I am pretty sure

**Reply by jdx** [reply]
That said, we probably shouldn't be doing any caching with autocomplete in most cases at least (like this one), but I actually am not sure how to get it to not cache lol

**Reply by syhol** [reply]
I've regenerated the mise completion file for fish and zsh. I've renamed the pdm.sh file to something else. I've disabled my fzf completions plugin. I've opened up new shells. and the behaviour still remains. 

To be clear I'm talking about shell completions. Actually in zsh, tasks with ':' were slightly broken for me, but I'll submit a separate ticket for that.

**Reply by syhol** [reply]
I can re-create it in a docker environment:
```console
❯ docker run --pull=always -it --rm --entrypoint bash jdxcode/mise:latest
root@88cc53e24167:/mise# echo '#!/usr/bin/env bash
> echo hi' >/mise/tasks/hi.sh
root@88cc53e24167:/mise# chmod +x tasks/hi.sh
root@88cc53e24167:/mise# apt update; apt install fish;
...
root@88cc53e24167:/mise# fish
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
root@88cc53e24167 /mise# cd
root@88cc53e24167 ~# mise use -g usage
root@88cc53e24167 ~# mise completions fish >~/.config/fish/completions/mise.fish
root@88cc53e24167 ~# echo "mise activate fish | source" >>~/.config/fish/config.fish 
root@88cc53e24167 ~# exec fish -l
root@88cc53e24167 ~# mise run <TAB>
```
Then  `hi.sh` will autocomplete.

**Reply by jdx** [reply]
ah I see what it is

**Reply by syhol** [reply]
Confirmed this works nicely now
